mtp_api/apps/transaction/tests/test_bank_admin_views.py

```diff
@@ -82,3 +85,294 @@ def test_create_list(self):
             ).count(),
             len(data_list)
         )
+
+    def _test_create_list_incomplete_data(self, null_field):
+        url = self._get_url()
+        data_list = self._get_transactions_data()
+
+        user = self.bank_admins[0]
+        data_list[0][null_field] = None
+
+        current_count = Transaction.objects.count()
+
+        response = self.client.post(
+            url, data=data_list, format='json',
+            HTTP_AUTHORIZATION=self.get_http_authorization_for_user(user)
+        )
+        self.assertEqual(response.status_code, http_status.HTTP_400_BAD_REQUEST)
+
+        # check no change in db
+        self.assertEqual(current_count, Transaction.objects.count())
+
+    def test_create_list_null_account_number_bad_request(self):
+        self._test_create_list_incomplete_data('sender_account_number')
+
+    def test_create_list_null_sort_code_bad_request(self):
+        self._test_create_list_incomplete_data('sender_sort_code')
+
+    def test_create_list_null_amount(self):
+        self._test_create_list_incomplete_data('amount')
```

Comment:
I'm not sure I agree that it's a single test with different inputs: it certainly appears that way now, but they're three distinct fields, and could all have different behaviours in future. The current tests imply that there's some sort of intentional link between the result of the three tests, when they actually just happen to have the same response to invalid input across three distinct fields.

If we were testing multiple invalid inputs to a single field, I'd probably agree that it's the same test (though I'd probably suggest using a [dataprovider-type pattern](https://pypi.python.org/pypi/unittest-data-provider/1.0.0) would be a clearer way to represent that).

I'd suggest that having a helper to null a given data list field is a more appropriate abstraction that reduces the duplication enough that doesn't imply intent in the result of the call to the API. The methods could also have the `url` and `user` objects inlined into the `self.client.post()` and be slightly less duplicate-y, I think?